runtime/datastructures/src/hook_list.rs

```diff
@@ -203,6 +203,13 @@ pub unsafe extern "C" fn printList(file: *mut FILE, list: *const List, unit: *co
   }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn list_foreach(list: *mut List, copy: extern fn(block: *mut K)) {
```

Comment:
"copy" seems overly specific here. "process" could be fine. The rust ForEach trait just calls the function argument "f"